pkg/resource/rule/hooks.go

```diff
@@ -0,0 +1,120 @@
+package rule
+
+import (
+	"context"
+	"fmt"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws-controllers-k8s/eventbridge-controller/apis/v1alpha1"
+	svcapitypes "github.com/aws-controllers-k8s/eventbridge-controller/apis/v1alpha1"
+)
+
+func validateRuleSpec(spec v1alpha1.RuleSpec) error {
+	if s := spec.State; s != nil {
+		if !(*s == "ENABLED" || *s == "DISABLED") {
```

Comment:
You could use the [`samber/lo.Contains`](https://github.com/samber/lo#contains) library function and the [`svcsdk.RuleState_Values` ](https://github.com/aws/aws-sdk-go/blob/4d8fb842b89848f744f8b194ff0ed8c2eff29091/service/eventbridge/api.go#L18010)function here:

```go
import (
        "github.com/samber/lo"
	svcsdk "github.com/aws/aws-sdk-go/aws/service/eventbridge"
)

var (
	errInvalidRuleState = fmt.Errorf(
		"invalid Spec: %q must be in %q",
		"spec.state", svcsdk.RuleState_Values(),
	)
)
...
		if s := spec.State; s != nil {
			if !lo.Contains(svcsdk.RuleState_Values(), *s) {
				return errInvalidRuleState
			}
		}
```

Reply:
Not sure I want to add a library for this, but using `svcsdk.RuleState_Values()` instead of hardcoded values SGTM. Will adjust.